Modules/System/Azure Queues/src/AzureStorageQueuesImpl.Codeunit.al

```diff
@@ -0,0 +1,293 @@
+/// <summary>
+/// Provides internal implementations of the helper functions for working with Azure Storage Queues
+/// Reference: https://docs.microsoft.com/en-us/rest/api/storageservices/queue-service-rest-api
+/// </summary>
+codeunit 50101 "Azure Storage Queues Impl."
+{
+    Access = Internal;
+
+    procedure CreateQueue(StorageAccountName: Text; Queue: Text): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpResponseMessage: HttpResponseMessage;
+        HttpContent: HttpContent;
+    begin
+        HttpClient.Put(GetQueueEndpoint(StorageAccountName, Queue), HttpContent, HttpResponseMessage);
+        exit(HttpResponseMessage.IsSuccessStatusCode);
+    end;
+
+    procedure ListQueues(StorageAccountName: Text): List of [Text]
+    var
+        HttpClient: HttpClient;
+        HttpResponseMessage: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        HttpClient.Get(GetAzureQueueServiceEndpoint(StorageAccountName) + '&comp=list', HttpResponseMessage);
+        if HttpResponseMessage.IsSuccessStatusCode then begin
```

Comment:
The begin-end is not strictly necessary. An if-then would be more in line with AL guidelines.